orchagent/neighorch.cpp

```diff
@@ -169,29 +169,29 @@ bool NeighOrch::hasNextHop(const NextHopKey &nexthop)
     return m_syncdNextHops.find(nexthop) != m_syncdNextHops.end();
 }
 
-bool NeighOrch::addNextHop(const IpAddress &ipAddress, const string &alias)
+bool NeighOrch::addNextHop(const NextHopKey &nh)
 {
     SWSS_LOG_ENTER();
 
     Port p;
-    if (!gPortsOrch->getPort(alias, p))
+    if (!gPortsOrch->getPort(nh.alias, p))
     {
         SWSS_LOG_ERROR("Neighbor %s seen on port %s which doesn't exist",
-                        ipAddress.to_string().c_str(), alias.c_str());
+                        nh.ip_address.to_string().c_str(), nh.alias.c_str());
         return false;
     }
     if (p.m_type == Port::SUBPORT)
     {
         if (!gPortsOrch->getPort(p.m_parent_port_id, p))
         {
             SWSS_LOG_ERROR("Neighbor %s seen on sub interface %s whose parent port doesn't exist",
-                            ipAddress.to_string().c_str(), alias.c_str());
+                            nh.ip_address.to_string().c_str(), nh.alias.c_str());
             return false;
         }
     }
 
-    NextHopKey nexthop = { ipAddress, alias };
-    if(m_intfsOrch->isRemoteSystemPortIntf(alias))
+    NextHopKey nexthop(nh);
```

Comment:
? Not sure I understood. Isn't "nh" already a NextHopKey object.

Reply:
@smaheshm: Yes, nh is already a NextHopKey, but compiler doesn't allow nh to be modified within this function, so a copy is made that can be modified.